docs/source/quicktour.mdx

```diff
@@ -99,12 +99,16 @@ The [`pipeline`] can also iterate over an entire dataset. Start by installing th
 pip install datasets 
 ```
 
-Create a [`pipeline`] with the task you want to solve for and the model you want to use. Set the `device` parameter to `0` to place the tensors on a CUDA device:
+Create a [`pipeline`] with the task you want to solve for and the model you want to use. Set the `device` parameter to `0` to place the tensors on a CUDA device if you
+it is available:
 
 ```py
+>>> import torch
 >>> from transformers import pipeline
 
->>> speech_recognizer = pipeline("automatic-speech-recognition", model="facebook/wav2vec2-base-960h", device=0)
+>>> device = 0 if torch.cuda.is_available() else -1
```

Comment:
This is the only time where we explicitly say in the doc tests that we should run on GPU. I think it's fine here to add an additional `if torch.cuda.is_availabsle()` to make it also pass on CPU. Ok for you @sgugger ?

Reply:
```suggestion
>>> device = "cuda" if torch.cuda.is_available() else "cpu"
```
This is clearer.

Reply:
Think this doesn't work no @Narsil ?

Reply:
Indeed. Let's remove GPU entirely then as it's not natural at all for a newcomer from PyTorch (@Narsil why doesn't `device` support the same thing as `torch.device`? This is very confusing...)

Reply:
But agree with @sgugger here that it should work. It's weird that `device` has to be an `int` here

Reply:
How, that is old code being moved over and over I think.
I guess initially it was done that way to be roughly agnostic to TF vs PT.

I do agree that being at least compatible with native things is OK (it should be at least).

For the example, I guess it depends where we are, but it's nice when the pipeline uses whatever backend is installed on your end (so not relying on `torch.device` nor `tf.device('/device:GPU:2')` by default but a simple subset for simple usage)

Reply:
Oh yes, I hadn't even realized we are putting some framework-specific code in the generic example. All the more reason to remove the `device` altogether, we shouldn't import `torch` here (or have some TF block that does the equivalent, but for this is starting to get too complex for the first part of the quicktour).

Reply:
Let's remove the `device` altogether ! Fine by me.